### DISCUSSION TITLE
Is it possible to export Pyvista Heatmap into CSV

### DISCUSSION THREAD

**Discussion by SofianeHamlaoui** [pyvista/pyvista#4590] [Q&A] [answered]
So, I tried getting a heatmap ( Elevation Heatmap ) using Pyvista of the Voxelized mesh I've had, and it looks like this : 

![image](https://github.com/pyvista/pyvista/assets/16967174/8a75514c-6682-482b-b312-449017ea83f3)

is it possible to export it to a CSV ?

**Comment by SofianeHamlaoui** [comment]
For the moment, I've got this using cell_centers, my goal is to only get the supperior cells 

![image](https://github.com/pyvista/pyvista/assets/16967174/46340104-16ff-403e-a45d-2f959e64a753)

So that I can use them, to get the heatmap and transfer it to a heightmap and export it to a CSV file

**Comment by NahomKidane** [comment, marked as answer]
You can access the scalars. Which is numpy and you can write it to csv.
a= mesh['Elevation'].
numpy.savetxt("foo.csv", a, delimiter=","). You also use pandas.

**Reply by SofianeHamlaoui** [reply]
do you think is possible to save to 32 x 32 CSV files ( jumps to a new file for each 32 x 32 ) ?
